TypeScript type error: missing rules_path in test config objects

## Description

After the `rules_path` feature was added to `LeonidasConfig` as a required field, several test files still have mock config objects without this property, causing `tsc --noEmit` to fail with `TS2345` errors.

## Error

```
error TS2345: Argument of type '{ label: string; model: string; ... }' is not assignable to parameter of type 'LeonidasConfig'.
  Property 'rules_path' is missing in type '...' but required in type 'LeonidasConfig'.
```

## Affected Files

- `src/main.test.ts` — 11 config objects missing `rules_path`
- `src/config.test.ts` — 1 config object missing `rules_path`

## Impact

- `npm run typecheck` fails (exit code 2)
- CI pipeline fails at type checking step

--- a/src/config.test.ts
+++ b/src/config.test.ts
@@ -100,6 +100,7 @@ describe("config", () => {
         ...defaultConfig,
         model: "claude-opus-4",
         max_turns: 100,
+        rules_path: ".github/leonidas-rules",
       });
     });
 

--- a/src/main.test.ts
+++ b/src/main.test.ts
@@ -69,6 +69,7 @@ describe("main", () => {
         allowed_tools: ["Read", "Write"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -123,6 +124,7 @@ describe("main", () => {
         allowed_tools: ["Read", "Write", "Edit"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -202,6 +204,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -253,6 +256,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -304,6 +308,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -390,6 +395,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -446,6 +452,7 @@ describe("main", () => {
         allowed_tools: ["Read", "Write"],
         max_turns: 100,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -524,6 +531,7 @@ describe("main", () => {
         allowed_tools: ["Read", "Write", "Edit", "Bash(git:*)"],
         max_turns: 60,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -581,6 +589,7 @@ describe("main", () => {
         allowed_tools: ["Read", "Write", "Edit", "Bash(npm:*)", "Bash(git:*)"],
         max_turns: 75,
         language: "ja",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -654,6 +663,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");
@@ -724,6 +734,7 @@ describe("main", () => {
         allowed_tools: ["Read"],
         max_turns: 50,
         language: "en",
+        rules_path: ".github/leonidas-rules",
       });
 
       const { buildSystemPrompt } = await import("./prompts/system");

